components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/CacheBackedActionMgtDAO.java

```diff
@@ -124,16 +126,45 @@ public Map<String, Integer> getActionsCountPerType(Integer tenantId) throws Acti
     }
 
     @Override
-    public Action getActionByActionId(String actionId, Integer tenantId) throws ActionMgtException {
+    public Action getActionByActionId(String actionType, String actionId, Integer tenantId) throws ActionMgtException {
 
-        return actionManagementDAO.getActionByActionId(actionId, tenantId);
+        ActionTypeCacheKey cacheKey = new ActionTypeCacheKey(actionType);
+        ActionCacheEntry entry = actionCacheByType.getValueFromCache(cacheKey, tenantId);
+        List<Action> actionsFromCache = new ArrayList<>();
```

Comment:
This is used only within 'updateCache'. So why not just defined within that method

Reply:
This is used in both checking action in cache with given id and same object pass to updateCache method.

Reply:
In line 138 you are reading from the cache entry so why the array list is needed

Reply:
Addressed with commit f73376335f5e1226d8f08d1c0d7f12b3a2340949